src/preferences/preferences.ts

```diff
@@ -0,0 +1,42 @@
+import { ipcRenderer } from 'electron';
+
+function switchTab(tab: string) {
+  ipcRenderer.send('switch-preferences-tab', tab);
+  updateMenuBar(tab);
+}
+
+function updateMenuBar(tab: string) {
+  // there must be anly one selected tab
+  let selectedDiv = document.getElementsByClassName("selected")[0];
+
+  // remove the selected class and add it to the new tab
+  selectedDiv.classList.remove("selected");
+  document.getElementById(tab).classList.add("selected");
+
+  // capitalize the first letter and update the title of the dialog
+  const upper = tab.charAt(0).toUpperCase() + tab.substring(1);
+  document.getElementById("window-title").textContent = upper;
+}
+
+// TODO @afiune implement this functionality
```

Comment:
IMO all these three TODOs should be done on a separate PR, if we want to use this PR I would be ok with it, just a heads-up that the PR could get a bit big since it is already `+1k` lines of code.